### DISCUSSION TITLE
AzerothCore on Windows with Codeblocks MinGW GCC Compiler + Cmake

### DISCUSSION THREAD

**Discussion by skxawng-cu** [azerothcore/azerothcore-wotlk#3961] [General]
Can I build AzerothCore on Windows with Codeblocks MinGW GCC Compiler + Cmake???

##### SMALL DESCRIPTION:

I have a slow internet connection and it is very difficult to download VisualStudio 2017 to build AzerothCore on Windows. I tried to compile it with the Codeblocks MinGW GCC compiler but I got many errors. 
I have always built AzerothCore on Debian 10, but I need to build it on Windows to make the development process more easy.

##### STEPS TO REPRODUCE THE PROBLEM:

1. Install all requirements to build AzerothCore on Windows but replace VS 2017 with Codeblocks MinGW GCC compiler (AC Wiki).
2. Follow the build instructions changing always VS 2017 with Codeblocks MinGW GCC compiler (AC Wiki).

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

a7970b711aa3bd3307b1f82302e84dce4f9ad011

##### OPERATING SYSTEM:

Windows 10 Pro x64

##### MODULES:

None

##### OTHER CUSTOMIZATIONS:

None

**Comment by Helias** [comment]
Hey, I always use Linux but I think that you can compile AC on Windows without Visual Studio, let's try it

**Comment by ghost** [comment]
You can also try downloading visual studio code and download the c++ compiler available on the extras, it's more customizable too

**Comment by ghost** [comment]
I've never tried anything but visual studio community but I can test tomorrow or so

**Comment by FrancescoBorzi** [comment]
Moved to the forum since this is not a bug report.

> but I need to build it on Windows to make the development process more easy.

what do you mean by that? I know this is opinion-based but I personally find Linux much more developer-friendly than Windows. Is there something specific that you need?

if you are looking for a C++ IDE that also runs on Linux, you can try CLion: https://github.com/azerothcore/forum/issues/4

**Comment by skxawng-cu** [comment]
> 
> 
> Moved to the forum since this is not a bug report.
> 
> > but I need to build it on Windows to make the development process more easy.
> 
> what do you mean by that? I know this is opinion-based but I personally find Linux much more developer-friendly than Windows. Is there something specific that you need?
> 
> if you are looking for a C++ IDE that also runs on Linux, you can try CLion: #4

I have Windows 10 Pro installed in my PC and when I go to build AC I create a virtual machine with Debian 10. I want to keep Win10 in my PC for now.
